### PR TITLE
[DPE-2079] Update rockcraft using Pebble services

### PR DESCRIPTION
**Pull request by deusebio**
Before merging, the workaround based on Docker should go away, when the [PR on Rockcraft](https://github.com/canonical/rockcraft/issues/235) is merged. 

Nevertheless the new image (which is also tested against integration tests) seems to be working properly

### CODE REVIEW

**Review by juditnovak — APPROVED** [review]
I kinda assume that this is what we need to do, and this is probably the best way :-)

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -37,8 +37,8 @@ jobs:
...
-      - name: Upload locally built ROCK artifact
-        uses: actions/upload-artifact@v3
-        with:
-          name: charmed-spark
-          path: "charmed-spark_*.rock"
+      # - name: Upload locally built ROCK artifact
+      #   uses: actions/upload-artifact@v3
+      #   with:
+      #     name: charmed-spark
+      #     path: "charmed-spark_*.rock"
```
Comment (deusebio) [inline .github/workflows/build.yaml]: **note** This is a step quite time consuming, but I don't see a reason to have this in, given in the integration tests we just upload things directly to the microk8s ctr

Can I just get rid of this? @taurus-forever

**Comment by deusebio** [comment]
The [PR on Rockcraft](https://github.com/canonical/rockcraft/issues/235) has been merged, and I have update this accordingly. The PR is now ready for review

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -37,8 +37,8 @@ jobs:
           tag=${version}-${base}_edge
           yq e -i ".version = \"$tag\"" rockcraft.yaml
           rockcraft pack --verbose
```
Comment (cjdcordeiro) [inline .github/workflows/build.yaml]: for future reference: there's a rockcraft action that might be useful: https://github.com/canonical/craft-actions/tree/main#rockcraft-pack

File: files/pebble/pebble-start.sh [hunk 3]
```diff
@@ -1,10 +1,8 @@
 #!/bin/bash
 
-# Start  Pebble
-/bin/pebble run --hold &
-
-# Wait Pebble to be up and running...
-sleep 1
+# This script still exists because passing of working directory to services has not yet landed
+# in Pebble (https://github.com/canonical/pebble/issues/158) and being in the write directory
+# is important for the entrypoint script
```
Comment (cjdcordeiro) [inline files/pebble/pebble-start.sh]: I'm not 100% sure that running pebble from within pebble is recommended, but in any case, you don't actually need to do that right? this file's last line could simply be: `cd /opt/spark && /bin/bash /opt/spark/entrypoint.sh $*`

File: rockcraft.yaml [hunk 4]
```diff
@@ -8,20 +8,25 @@ base: ubuntu:22.04
...
+      SPARK_HOME: /opt/spark
+      SPARK_CONFS: /etc/spark8t/conf
+      JAVA_HOME: /usr/lib/jvm/java-11-openjdk-amd64
+      PYTHONPATH: /opt/spark/python:/opt/spark8t/python/dist:/usr/lib/python3/dist-packages
+      PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/spark:/opt/spark/bin:/opt/spark/python/bin:/opt/spark-client/python/bin
+      HOME: /home/spark
+      KUBECONFIG: /home/spark/.kube/config
+      SPARK_USER_DATA: /home/spark
+    on-success: shutdown
+    on-failure: shutdown
```
Comment (cjdcordeiro) [inline rockcraft.yaml:27]: considering my comment in https://github.com/canonical/charmed-spark-rock/pull/23/files#r1229032686, if you add `user` and `group` here, then your entrypoint.sh could possibly be reduced to a single line

**Review by cjdcordeiro — APPROVED** [review]
nice! I left a few comments that aren't really blocking this from being merged, but rather optimization hints. So it LGTM ;)

File: .github/workflows/build.yaml (hunk 1 above)
Comment (taurus-forever) [reply .github/workflows/build.yaml]: You need it on the "release/publish" [stage](https://github.com/canonical/charmed-spark-rock/blob/3.3/edge/.github/workflows/publish.yaml#LL30C1-L33C1).

**Review by taurus-forever — CHANGES_REQUESTED** [review]
IMHO, you still need artifacts upload.

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -23,22 +23,11 @@ jobs:
...
-          tag=${version}-${base}_edge
-          yq e -i ".version = \"$tag\"" rockcraft.yaml
-          rockcraft pack --verbose
-      # - name: Upload locally built ROCK artifact
-      #   uses: actions/upload-artifact@v3
-      #   with:
-      #     name: charmed-spark
-      #     path: "charmed-spark_*.rock"
+        uses: canonical/craft-actions/rockcraft-pack@main
+        id: rockcraft
```
Comment (taurus-forever) [inline .github/workflows/build.yaml:28]: Previously Jarred was changing version inside rockcraft.yaml:
```
yq e -i ".version = \"$tag\"" rockcraft.yaml
```
Are you sure you will have the same result/tag now?
LGTM, if so.

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**